src/enzo/CallInSitulibyt.C

```diff
@@ -0,0 +1,452 @@
+    params->mass_unit = DensityUnits * LengthUnits * LengthUnits * LengthUnits; /* this right? */
+    params->time_unit = TimeUnits;
+    params->velocity_unit = VelocityUnits;
+    params->magnetic_unit = sqrt(4.0 * 3.141592653589793238462643383279502884L * DensityUnits) * VelocityUnits;
+    params->periodicity[0] = (MetaData->LeftFaceBoundaryCondition[0] == periodic);
+    params->periodicity[1] = (MetaData->LeftFaceBoundaryCondition[1] == periodic);
+    params->periodicity[2] = (MetaData->LeftFaceBoundaryCondition[2] == periodic);
+    params->dimensionality = MetaData->TopGridRank;
+    params->domain_dimensions[0] = MetaData->TopGridDims[0];
+    params->domain_dimensions[1] = MetaData->TopGridDims[1];
+    params->domain_dimensions[2] = MetaData->TopGridDims[2];
+	for (int d = 2; d > params->dimensionality - 1; d--) {
+		params->domain_dimensions[d] = 1;
+	}
+
+    params->refine_by = RefineBy;
+    params->index_offset = 1;
+    params->num_grids = num_grids;
+    params->num_grids_local = num_local_grids;
+    /* We do things by DataLabel */
+
+    for (i = 0; i < MAX_NUMBER_OF_BARYON_FIELDS; i++) {
+        if (!DataLabel[i]) {
+            params->num_fields = i;
+            break;
+        }
+    }
+
+    // Add fields for Temperature/Cooling_Time derived field from enzo
+    params->num_fields += 2;
+
+    // Add active particle ptypes
+    params->num_par_types = 1 + EnabledActiveParticlesCount; // DarkMatter and Other ActiveParticle (ex: SmartStar)
+    yt_par_type *par_type_list = new yt_par_type [params->num_par_types];
+
+    par_type_list[0].par_type = "DarkMatter";
+    par_type_list[0].num_attr = 3 + 3 + 1 + 1 + 1 + NumberOfParticleAttributes;
+
+    // the attributes name should be alive within the entire libyt in situ analysis,
+    // because libyt does not make a copy of the names.
+    std::vector<std::vector<std::string>> active_particles_attributes;
+    std::vector<std::vector<hid_t>> active_particles_attributes_hdf5type;
+
+    for (int i = 0; i < EnabledActiveParticlesCount; i++) {
+        ActiveParticleType_info *ActiveParticleTypeToEvaluate = EnabledActiveParticles[i];
+        active_particles_attributes.emplace_back(ActiveParticleTypeToEvaluate->GetParticleAttributeNames());
+        active_particles_attributes_hdf5type.emplace_back(ActiveParticleTypeToEvaluate->GetParticleAttributesHDF5DataType());
+
+        par_type_list[1 + i].par_type = ActiveParticleTypeToEvaluate->particle_name.c_str();
+        par_type_list[1 + i].num_attr = active_particles_attributes[i].size();
+    }
+
+    params->par_type_list = par_type_list;
+
+    if (yt_set_Parameters(params) != YT_SUCCESS){
+        fprintf(stderr, "Error in libyt API yt_set_Parameters\n");
+        return FAIL;
+    }
+
+    delete [] par_type_list;
+
+    yt_particle *particle_list;
+    yt_get_ParticlesPtr(&particle_list);
+
+    // TODO: make sure enzo's particle is always DarkMatter
+    particle_list[0].par_type = "DarkMatter";
```

Comment:
I'm confused why there is one and only one hard-coded DarkMatter particle.

Reply:
I will fix this.
Could you point me to the Enzo documents about how each particle types are handled or named?